velox/dwio/common/tests/DataSinkTest.cpp

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "velox/dwio/common/DataSink.h"
+#include "velox/common/base/Fs.h"
+#include "velox/exec/tests/utils/TempDirectoryPath.h"
+
+#include <boost/uuid/uuid_generators.hpp>
+#include <boost/uuid/uuid_io.hpp>
+#include <gmock/gmock.h>
+#include <gtest/gtest.h>
+
+using namespace ::testing;
+using namespace facebook::velox::exec::test;
+
+namespace facebook::velox::dwio::common {
+
+TEST(LocalFileSinkTests, create) {
+  LocalFileSink::registerFactory();
+
+  auto root = fs::path(TempDirectoryPath::create()->path);
+  auto subdir = fs::path();
+  for (auto i = 0; i < 4; i++) {
+    subdir =
+        subdir / boost::uuids::to_string(boost::uuids::random_generator()());
```

Comment:
The test would be more readable if only top-level directory name is auto-generated, but the rest of the directories and the file name are hard-coded.

Reply:
I see. Revised.